syntax = "proto3";

package server;

import "google/protobuf/empty.proto";

message Status {
    bool ok = 1;
}

message WorkerId {
    uint32 id = 1;
}

message WorkerList {
    repeated uint32 id = 1;
}

message WorkType {
    bytes map_function_pickled = 1;
    bytes reduce_function_pickled = 2;
    repeated string args = 3;
}


message JobId {
    string id = 1;
}

message NumberOfWorkers {
    uint32 num = 1;
}

// Worker service
service WorkerManagement {
    rpc create (NumberOfWorkers) returns (Status) {}
    rpc list (google.protobuf.Empty) returns (WorkerList) {}
    rpc delete (WorkerId) returns (Status) {}
    rpc job (WorkType) returns (JobId) {}
}

